keras_nlp/models/generative_task.py

```diff
@@ -137,25 +183,19 @@ def generate(
         """Generate text given prompt `inputs`.
 
         This method generates text based on given `inputs`. The sampling method
-        used for generation can be set via the `compile()` method.
+        used for generation can be set in the `compile` method.
 
         If `inputs` are a `tf.data.Dataset`, outputs will be generated
         "batch-by-batch" and concatenated. Otherwise, all inputs will be handled
         as a single batch.
 
-        If a `preprocessor` is attached to the model, `inputs` will be
-        preprocessed inside the `generate()` function and should match the
-        structure expected by the `preprocessor` layer (usually raw strings).
-        If a `preprocessor` is not attached, inputs should match the structure
-        expected by the `backbone`. See the example usage above for a
-        demonstration of each.
+        If a `preprocessor` is attached to the model, `inputs` should be
+        strings and returned sequences will be strings. Otherwise, inputs should
+        be preprocessed before calling `generate()`, and returned sequences will
+        be token ids.
 
         Args:
-            inputs: python data, tensor data, or a `tf.data.Dataset`. If a
```

Comment:
Just checking -- is cutting the detail out of this docstring intentional? (I assume so since the same info is repeated above)

Reply:
Ah no this was an overaggressive rebase, will keep the old docstring.